### PR TITLE
docker.yaml: Split `builder` image generation to another workflow job

### PR DESCRIPTION
**Pull request by hjpotter92**
- Cleanup hosted runner further as suggested on discord [1]

[1]: https://discord.com/channels/423160867534929930/750796877762527262/1138806765648105544

### CODE REVIEW

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -89,6 +84,79 @@ jobs:
             type=raw,value=${{ github.event.pull_request.head.ref }}
             type=raw,value=stable,enable=${{ startsWith(github.event.ref, 'refs/tags/v') }}
 
+      - name: Build and push livepeer docker image
+        uses: docker/build-push-action@v4
+        with:
+          build-args: |
+            BUILD_TAGS=${{ steps.build-tag.outputs.build-tags }}
+          # linux/arm64 is disabled until
```
Comment (leszko) [inline .github/workflows/docker.yaml]: The related ticket is resolved, should we remove this comment?

File: .github/workflows/docker.yaml [hunk 2]
```diff
@@ -89,6 +84,79 @@ jobs:
...
+
+      - name: Cleanup hosted runner
+        run: |
+          sudo apt purge -yqq dotnet-* mono-* llvm-* adoptopenjdk-* libllvm* powershell* openjdk-* \
+          temurin-* mongodb-* firefox mysql-* \
+          hhvm google-chrome-stable \
+          libgl1-mesa-dri microsoft-edge-stable \
+          google-cloud-sdk azure-cli
+          sudo apt autoremove -y
+          sudo rm -rf /usr/share/dotnet /usr/local/lib/android
```
Comment (leszko) [inline .github/workflows/docker.yaml:125]: Do we need to remove it both here and in line 40?

**Review by leszko — APPROVED** [review]

File: .github/workflows/docker.yaml (hunk 2 above)
Comment (hjpotter92) [reply .github/workflows/docker.yaml:125]: not really! but just keeping it here for future-proofing against having similar issues (out of disk space on github runner). go-livepeer won't be needing android libraries (and other packages being removed).

File: .github/workflows/docker.yaml (hunk 1 above)
Comment (hjpotter92) [reply .github/workflows/docker.yaml]: you're right. updated.

**Pull request merged**